### ISSUE TITLE
Scroll position not saved when leaving and returning to home tab

### ISSUE THREAD

**Issue by kevdesilva01** [UCSD/campus-mobile#1651]
Time Tracking: MA-34

When going into the parking card or NTPLLN card, the user is redirected to other screens. Once that user is done selecting something in those screens, the NTPLLN Card automatically routes back to the home screen and the parking card routes to the home screen after a user clicks the done button (v2). When it reroutes back to the home screen, the user should be taken back to the position where they left off instead of the very top. 

This is done when:
- [x] Identify the areas in the app which cause the Home view to scroll to the top of the screen after clicking back or '<'
- [x] A users position on the home screen is saved when leaving and the user re-enters the home screen at the same place 

- [x] [PR 1759 Test Plan](https://ucsdcloud.sharepoint.com/sites/WorkplaceTechnologyServices-CampusMobileBuilds/Shared%20Documents/Campus%20Mobile%20Builds/Pull%20Request%20Testing/PR-1759-Test-Plan-7.18.0-QA-2091.xlsx?web=1) PASS on iOS
- [x] [PR 1759 Test Plan](https://ucsdcloud.sharepoint.com/sites/WorkplaceTechnologyServices-CampusMobileBuilds/Shared%20Documents/Campus%20Mobile%20Builds/Pull%20Request%20Testing/PR-1759-Test-Plan-7.18.0-QA-2092.xlsx?web=1) PASS on Android

**Comment by gmihir** [comment]
The current PR is not consistent in saving the user position. It only worked 1/3 times in the video below

**Comment by gmihir** [comment]
https://user-images.githubusercontent.com/13931192/144344569-855bf052-6622-4bc8-8b63-a07aede4d664.MOV

**Comment by xxwwyytt1** [comment]
problem was the state of home tab gets destroyed when switching to another tab, resolved with AutomaticKeepAliveClientMixin implementation. But the scroll position saved was incorrect when switching back to home tab from other tabs. Doubting if it's synchronization issue.

**Comment by kevdesilva01** [comment]
Android testing passed. Tested extensively on parking card. The only small thing I noticed was that when pressing the done button from the "Neighborhoods" screens, it doesn't go back to the home tab. It goes to the Manage Lots screen again. The other two options (Parking Structure) and Parking lots, go to the home menu. This might have something to do with the implementation of the "pop" function, as the two screens that work as intended currently need to be popped twice to return to the home tab, but the Neighborhoods screen would need to be popped 3 times.

**Comment by c3bryant** [comment]
Encountered a bug in 7.19 testing.

Student logout is not working as expected.

Upon student logout, the user is not fully logged out. The Profile tab shows 'Logged in' with an empty username. The Home view still shows some authenticated web cards

**Comment by c3bryant** [comment]
woznaut reported an issue on iOS, confirmed on Android.

Bug: The Cards view under the Profile tab is not functional. There are two "No internet" popups when navigating to this view. When the popups are dismissed, the view is blank.

Reverting PR 1935 to investigate a fix.

**Comment by tkiyohar** [comment]
> woznaut reported an issue on iOS, confirmed on Android.
> 
> Bug: The Cards view under the Profile tab is not functional. There are two "No internet" popups when navigating to this view. When the popups are dismissed, the view is blank.
> 
> Reverting PR 1935 to investigate a fix.

I will be investigating this error in the coming weeks.

**Comment by tkiyohar** [comment]
All issues regarding this PR have been fixed and the code for this PR has been ported over to the newest experimental version of the mobile app on a brand-new PR (please see: https://github.com/UCSD/campus-mobile/pull/2019).

**Comment by c3bryant** [comment]
Testers Needed: Please complete updated *PR testing* section in iOS and Android test plans linked above.
